src/status.rs

```diff
@@ -0,0 +1,126 @@
+use std::fmt;
+use std::process::ExitCode;
+
+use colored::Colorize;
+
+use crate::nixpkgs_problem::NixpkgsProblem;
+
+pub enum Status {
+    /// It's all green.
+    ValidatedSuccessfully,
+
+    /// The base branch is broken, but this PR fixes it. Nice job!
+    BranchHealed,
+
+    /// The base branch fails, the PR doesn't fix it, and the PR may also introduce additional
+    /// problems.
+    BranchStillBroken(Vec<NixpkgsProblem>),
+
+    /// This PR introduces the problems listed. Please fix them before merging, otherwise the base
+    /// branch would break.
+    ProblemsIntroduced(Vec<NixpkgsProblem>),
+
+    /// This PR introduces additional instances of discouraged patterns. Merging is discouraged but
+    /// would not break the base branch.
+    DiscouragedPatternedIntroduced(Vec<NixpkgsProblem>),
+
+    /// Some other error occurred.
+    Error(anyhow::Error),
+}
+
+impl Status {
+    fn fmt(&self, f: &mut fmt::Formatter, use_color: bool) -> fmt::Result {
+        let maybe_green = |s: &str| if use_color { s.green() } else { s.normal() };
+        let maybe_yellow = |s: &str| if use_color { s.yellow() } else { s.normal() };
+        let maybe_red = |s: &str| if use_color { s.red() } else { s.normal() };
+
+        let errors = match self {
```

Comment:
```suggestion
        let (errors, message) = match self {
```

How about this to avoid the `unreachable!`

Reply:
I figured out a way to avoid it using a getter for errors.